test/grisp_connect_api_SUITE.erl

```diff
@@ -61,20 +58,49 @@ auto_connect_test(_) ->
     ?assertMatch(ok, wait_connection()).
 
 ping_test(_) ->
-    ?assertMatch({ok, <<"pang">>}, grisp_connect:ping()).
+    {Pid, MRef} = ?asyncAssertEqual({ok, <<"pong">>}, grisp_connect, ping, []),
+    Id = ?receiveJsonRpcRequest(<<"post">>, #{type := <<"ping">>}),
```

Comment:
Some test may requires to get some info out of the message to build the response, so it would probably be better to have a generic assert like:
```
#{id := Id} = ?assertJsonRpcMessage(#{id:= _, method := <<"post">>, params := #{type := <<"ping">>})
```
But maybe we could have both to make simple tests less verbose....
I would still have the macro start with `?assertXXXX` though as it contains an assert...

Reply:
I agree on returning the whole map. I will change that. I disagree on the naming. It is already a very long macro name, but I want "receive", "json-rpc" and "request" in there, because that is what I checking for. (The macro doesn't work for notifications, we can add one, when we need one.)

Reply:
Maybe "JsonRpc" isn't that important in the macro name?

Reply:
I would be happy with `?assertRequest` :)
Why is the function not working for notifications ?

Reply:
Notifications don't have an `id` and I check for it in the pattern match.

Reply:
I understand so that make the test less verbose. We could then have (some implemented later):